### PR TITLE
Fix minor bug in volume_clipping.py example

### PR DESCRIPTION
**Pull request by kevinyamauchi**
First of all, thanks for an amazing project!

Pressing a modifier (e.g., shift) in the `examples/scene/volume_clipping.py` example caused an error (traceback below) since `''in 'xyzo'` evaluates to `True` [here](https://github.com/vispy/vispy/blob/3ef02c5f1ae6bf34616060e350d7e1818c8d3c21/examples/scene/volume_clipping.py#L117). I have added a small check in `add_clip()` to make sure `mode` is valid before attempting to add a clipping plane.

edit: I also removed some lines of code that seemed unused.

```python
WARNING: Traceback (most recent call last):
  File "/Users/kyamauch/Documents/vispy/examples/scene/volume_clipping.py", line 130, in <module>
    app.run()
  File "/Users/kyamauch/Documents/vispy/vispy/app/_default_app.py", line 60, in run
    return default_app.run()
  File "/Users/kyamauch/Documents/vispy/vispy/app/application.py", line 163, in run
    return self._backend._vispy_run()
  File "/Users/kyamauch/Documents/vispy/vispy/app/backends/_qt.py", line 292, in _vispy_run
    return app.exec_()
  File "/Users/kyamauch/Documents/vispy/vispy/app/backends/_qt.py", line 525, in event
    out = super(QtBaseCanvasBackend, self).event(ev)
  File "/Users/kyamauch/Documents/vispy/vispy/app/backends/_qt.py", line 519, in keyPressEvent
    self._keyEvent(self._vispy_canvas.events.key_press, ev)
  File "/Users/kyamauch/Documents/vispy/vispy/app/backends/_qt.py", line 571, in _keyEvent
    func(native=ev, key=key, text=str(ev.text()), modifiers=mod)
  File "/Users/kyamauch/Documents/vispy/vispy/util/event.py", line 453, in __call__
    self._invoke_callback(cb, event)
  File "/Users/kyamauch/Documents/vispy/vispy/util/event.py", line 471, in _invoke_callback
    _handle_exception(self.ignore_callback_errors,
  << caught exception here: >>
  File "/Users/kyamauch/Documents/vispy/vispy/util/event.py", line 469, in _invoke_callback
    cb(event)
  File "/Users/kyamauch/Documents/vispy/examples/scene/volume_clipping.py", line 118, in on_key_press
    add_clip(volume, event.text)
  File "/Users/kyamauch/Documents/vispy/examples/scene/volume_clipping.py", line 102, in add_clip
    new_plane = clip_modes[mode]
KeyError: ''
ERROR: Invoking <function on_key_press at 0x7fd0e6fbb1f0> for KeyEvent
```

### CODE REVIEW

File: examples/scene/volume_clipping.py [hunk 1]
```diff
@@ -50,30 +46,6 @@
 axis.transform = affine
 
 
-# create colormaps that work well for translucent and additive volume rendering
```
Comment (kevinyamauchi) [inline examples/scene/volume_clipping.py:53]: @brisvag , is it okay to remove these lines?

Comment (brisvag) [reply examples/scene/volume_clipping.py:53]: Seems like it! Leftovers from my ugly copy-paste :P

**Review by brisvag — APPROVED** [review]
Thanks, I did not bother debugging this issue cause I saw it elsewhere and then forgot about it. It might be worth checking @alisterburt 's plane rendering example, I think it had the same issue!

File: examples/scene/volume_clipping.py [hunk 2]
```diff
@@ -99,11 +71,12 @@ def on_mouse_move(event):
...
-    if vol.clipping_planes is None:
-        vol.clipping_planes = new_plane
-    else:
-        vol.clipping_planes = np.concatenate([vol.clipping_planes, new_plane])
+    if mode in clip_modes:
+        new_plane = clip_modes[mode]
+        if vol.clipping_planes is None:
+            vol.clipping_planes = new_plane
+        else:
+            vol.clipping_planes = np.concatenate([vol.clipping_planes, new_plane])
```
Comment (djhoese) [inline examples/scene/volume_clipping.py:79]:
```suggestion
    if mode not in clip_modes:
        return
    new_plane = clip_modes[mode]
    if vol.clipping_planes is None:
        vol.clipping_planes = new_plane
    else:
        vol.clipping_planes = np.concatenate([vol.clipping_planes, new_plane])
```

Small nitpick.

**Review by djhoese — CHANGES_REQUESTED** [review]
Thanks for this. I honestly thought the colormap stuff was used when reviewing the example. I had one small change that I requested. Otherwise does the volume plane example need updating for this issue too?

**Comment by kevinyamauchi** [comment]
Thanks for the suggestion, @djhoese ! I've committed it and I will have a look if the plane rendering example needs similar cleanup.

**Comment by kevinyamauchi** [comment]
I just had a look and I don't think the volume_plane.py example has the issues fixed in this PR.

**Review by djhoese — APPROVED** [review]
Great! Thank you!

**Pull request merged**